_source/_posts/2022-02-28-shopping-app-with-node-minio.md

```diff
@@ -0,0 +1,706 @@
+---
+layout: blog_post
+title: Build a Basic Shopping App with Node.js and MinIO
+author: deepa-mahalingam
+by: internal-contributor
+communities: [javascript]
+description: "A tutorial that shows you how to build a basic shopping app with Node.js and MinIO."
+tags: [sign-in-widget, okta, node, minio]
+type: conversion
+image: blog/shopping-app-with-node-minio/cover-image.jpg
+github: https://github.com/oktadev/okta-express-minio-app-example
+tweets:
+- "Learn how to build a basic shopping app with @nodejs and @Minio 🍿"
+---
+
+These days, a lot of e-commerce applications use Node.js. Typically, e-commerce applications have many images and video assets managed in object storage. As the number of products in an e-commerce store increases, so does the size of the object storage. We'll build our Node.js application with a Kubernetes-friendly object storage service, MinIO, to scale with demand.
+
+This tutorial will guide you through the code to build a simple Node.js Shopping App with the Okta's Sign-In Widget for authentication and a [MinIO Server](https://min.io/) to host the store's assets. We will use Express as our application framework and Handlebars as the view engine. We will use the [MinIO Javascript Client SDK](https://docs.minio.io/docs/javascript-client-quickstart-guide) to fetch the application's image assets from the Minio Server.
+
+{% img
+blog/shopping-app-with-node-minio/home.png
+alt:"The example application"
+width:"100%" %}
```

Comment:
We typically use `width:800` and then center the image by adding `{: .center-image }`. See [our blog markdown conventions](https://github.com/oktadev/okta-blog#blog-markdown-conventions) for more information.